prow/plank/reconciler.go

```diff
@@ -944,3 +975,26 @@ func isRequestError(err error) bool {
 	}
 	return 400 <= code && code < 500
 }
+
+func countPendingOrOlderMatchingPJs(pj prowv1.ProwJob, pjs []prowv1.ProwJob) int {
+	var pendingOrOlderMatchingPJs int
+
+	for _, foundPJ := range pjs {
+		// Ignore self here.
+		if foundPJ.UID == pj.UID {
+			continue
+		}
+		if foundPJ.Status.State == prowv1.PendingState {
+			pendingOrOlderMatchingPJs++
+			continue
+		}
+
+		// At this point we know that foundPJ is in Triggered state, if its older than our prowJobs it gets
+		// priorized to make sure we execute jobs in creation order.
+		if foundPJ.CreationTimestamp.Before(&pj.CreationTimestamp) {
+			pendingOrOlderMatchingPJs++
```

Comment:
Do we want to break ties on timestamp ordering with lexicographic sort of UID or something?

Reply:
It does not matter to me and it was not the case for the `MaxConcurrency` mechanism.